dotCMS/src/main/java/com/dotcms/rendering/velocity/servlet/VelocityPreviewServlet.java

```diff
@@ -0,0 +1,359 @@
+        }
+    }
+
+
+    protected void doEditMode(HttpServletRequest request, HttpServletResponse response) throws Exception {
+        PageMode mode = PageMode.PREVIEW_MODE;
+        String uri = request.getRequestURI();
+
+        Host host = hostWebAPI.getCurrentHost(request);
+
+
+        // Getting the user to check the permissions
+        User user = com.liferay.portal.util.PortalUtil.getUser(request);
+
+
+        // Getting the identifier from the uri
+        Identifier id = APILocator.getIdentifierAPI()
+            .find(host, uri);
+        request.setAttribute("idInode", String.valueOf(id.getId()));
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY HTML INODE=" + id.getId());
+
+        Template template;
+        Template hostVariablesTemplate = null;
+
+        // creates the context where to place the variables
+        response.setContentType(CHARSET);
+        request.setAttribute("EDIT_MODE", Boolean.TRUE);
+        Context context = VelocityUtil.getWebContext(request, response);
+
+        IHTMLPage htmlPage = VelocityUtil.getPage(id, request, false, context);
+
+        context.put("dotPageContent", new ContentMap(((Contentlet) htmlPage), user, mode, host, context));
+
+
+        com.dotmarketing.portlets.templates.model.Template cmsTemplate = APILocator.getHTMLPageAssetAPI()
+            .getTemplate(htmlPage, true);
+
+        if (cmsTemplate == null) {// DOTCMS-4051
+            cmsTemplate = new com.dotmarketing.portlets.templates.model.Template();
+            Logger.debug(VelocityPreviewServlet.class, "HTMLPAGE TEMPLATE NOT FOUND");
+        }
+
+        Identifier templateIdentifier = APILocator.getIdentifierAPI()
+            .find(cmsTemplate);
+
+        Logger.debug(VelocityPreviewServlet.class, "VELOCITY TEMPLATE INODE=" + cmsTemplate.getInode());
+
+        VelocityUtil.makeBackendContext(context, htmlPage, cmsTemplate.getInode(), id.getURI(), request, mode, host);
+
+
+        PageContextBuilder builder = new PageContextBuilder(htmlPage, user, PageMode.PREVIEW_MODE);
+        builder.addAll(context);
+
+        hostVariablesTemplate = VelocityUtil.getEngine()
+            .getTemplate(PageMode.EDIT_MODE.name() + File.separator + host.getIdentifier() + "." + VelocityType.SITE.fileExtension);
+
+        if (cmsTemplate.isDrawed()) {// We have a designed template
+            // Setting some theme variables
+            Map<String, Object> dotThemeData = DotTemplateTool.theme(cmsTemplate.getTheme(), host.getIdentifier());
+            context.put("dotTheme", dotThemeData);
+            context.put("dotThemeLayout", DotTemplateTool.themeLayout(cmsTemplate.getInode()));
+            // Our designed template
+            template = VelocityUtil.getEngine()
+                .getTemplate((String) dotThemeData.get("templatePath"));
+        } else {
+            template = VelocityUtil.getEngine()
+                .getTemplate(PageMode.EDIT_MODE.name() + File.separator + templateIdentifier.getId() + "."
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "getTemplate". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Add a "try/catch" block for "getTemplate". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1989)